drivers/clk/renesas/rzg2l-cpg.c

```diff
@@ -158,6 +164,46 @@ struct rzg2l_cpg_priv {
 	struct rzg2l_pll5_mux_dsi_div_param mux_dsi_div_params;
 };
 
+/**
+ * get_dsi_status - Retrieves the status of the DSI node from the device tree.
+ * @dev: Pointer to the device structure.
+ *
+ * This function searches the device tree for the DSI node identified by the
+ * compatible string "renesas,r9a07g044-mipi-dsi" and reads its "status"
+ * property. It determines if the DSI node is enabled or disabled.
+ *
+ * Return: 1 if the DSI node is enabled, 0 if it is disabled,
+ *         -ENODEV if the DSI node is not found,
+ *         -EINVAL if the "status" property is missing.
+ */
+static int get_dsi_status(const char *node_compatible, struct device_node **dsi_node)
+{
+    const char *status;
+
+    /* Get the DSI node from the device tree */
+    *dsi_node = of_find_compatible_node(NULL, NULL, node_compatible);
+    if (!dsi_node) {
+        pr_err("DSI node not found\n");
+        return -ENODEV;
+    }
+
+    /* Read the status property from the DSI node */
+    status = of_get_property(*dsi_node, "status", NULL);
+    if (!status) {
+        pr_err("DSI node does not have a status property\n");
+        return -EINVAL;
+    }
+
+    /* Check if the DSI node status is "disabled" */
+    if (strcmp(status, "disabled") == 0) {
+        printk("DSI is disabled\n");
+        return 0;
+    } else {
+        printk("DSI is enabled\n");
+        return 1;
+    }
+}
+
```

Comment:
@long-trinh-rvc ,

There are 2 more points regarding this function.
- Change the function name to something more generic, such as `get_node_status`, since it is no longer specific to DSI, and also modify the comment.
- Remove printk, as it's not needed.